utilities/llmd_utils.py

```
@@ -0,0 +1,632 @@
+    timeout: int = 300,
+    teardown: bool = True,
+) -> Generator[Gateway, None, None]:
+    """
+    Context manager to create and manage a Gateway resource using ocp_resources.
+
+    Args:
+        client: Kubernetes dynamic client
+        name: Gateway name
+        namespace: Gateway namespace
+        gateway_class_name: The name of the GatewayClass resource
+        listeners: List of listener configurations
+        infrastructure: Infrastructure configuration
+        wait_for_condition: Whether to wait for the gateway to be programmed
+        timeout: Timeout in seconds for waiting
+        teardown: Whether to clean up the resource
+
+    Yields:
+        Gateway: The created Gateway resource
+    """
+    if listeners is None:
+        listeners = [
+            {
+                "name": "http",
+                "port": 80,
+                "protocol": "HTTP",
+                "allowedRoutes": {"namespaces": {"from": "All"}},
+            }
+        ]
+
+    if infrastructure is None:
+        infrastructure = {"labels": {"serving.kserve.io/gateway": "kserve-ingress-gateway"}}
+    try:
+        existing_gateway = Gateway(
+            client=client,
+            name=name,
+            namespace=namespace,
+            api_group="gateway.networking.k8s.io",
+        )
+        if existing_gateway.exists:
+            LOGGER.info(f"Cleaning up existing Gateway {name} in namespace {namespace}")
+            existing_gateway.delete(wait=True, timeout=Timeout.TIMEOUT_2MIN)
+    except Exception as e:
+        LOGGER.debug(f"No existing Gateway to clean up: {e}")
+    gateway_body = {
+        "apiVersion": "gateway.networking.k8s.io/v1",
+        "kind": "Gateway",
+        "metadata": {
+            "name": name,
+            "namespace": namespace,
+        },
+        "spec": {
+            "gatewayClassName": gateway_class_name,
+            "listeners": listeners,
+            "infrastructure": infrastructure,
+        },
+    }
```

Do you need this? IMO we should assume that the cluster is in a good state before running tests, and I think deleting existing gateway before running the tests could potentially mask failures, wdyt?